Makefile

```diff
@@ -39,6 +39,10 @@ lint: format fix clippy ## Runs all linting tasks at once (Clippy, fixing, forma
 .PHONY: doc
 doc: ## Generates & checks documentation
 	$(WARNINGS) cargo doc --all-features --keep-going --release --locked
+	docker run --rm \
+      -v $(PWD)/docs:/out \
+      -v $(PWD)/proto:/protos \
+      pseudomuto/protoc-gen-doc --doc_opt=/out/api.tmpl,api.md
```

Comment:
We probably need a separate CI check to ensure the generated docs match this?

I'm also unsure if this should be part of `make doc` like this; though maybe its fine.

I would:

- add a `README.md` with instructions to the `docs` folder with these instructions
- optionally remove these instructions from here (only manual invocation)
- add ci check to ensure we stay in sync